src/main/kotlin/com/cosmotech/api/containerregistry/ContainerRegistryService.kt

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.containerregistry
+
+import com.cosmotech.api.config.CsmPlatformProperties
+import com.cosmotech.api.exceptions.CsmClientException
+import com.cosmotech.api.exceptions.CsmResourceNotFoundException
+import java.util.Base64
+import org.apache.http.HttpStatus
+import org.apache.http.client.methods.HttpUriRequest
+import org.apache.http.client.methods.RequestBuilder
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+import org.json.JSONArray
+import org.json.JSONObject
+import org.springframework.http.HttpHeaders
+import org.springframework.stereotype.Service
+
+@Service("csmContainerRegistry")
+class ContainerRegistryService(private val csmPlatformProperties: CsmPlatformProperties) {
+  fun getEndpoint() = csmPlatformProperties.containerRegistry.registryUrl
+
+  fun getCredentials(user: String, password: String) =
+      Base64.getEncoder().encodeToString("$user:$password".toByteArray())
+
+  fun getHeaderAuthorization() =
+      "Basic " +
+          getCredentials(
+              csmPlatformProperties.containerRegistry.registryUserName!!,
+              csmPlatformProperties.containerRegistry.registryPassword!!)
+
+  private fun buildHttpRequest(repository: String): HttpUriRequest {
+    var reqBuilder = RequestBuilder.get()
+    reqBuilder = reqBuilder.setUri(getImageRegistryUri(repository))
+    reqBuilder = reqBuilder.addHeader(HttpHeaders.AUTHORIZATION, getHeaderAuthorization())
+
+    return reqBuilder.build()
+  }
+
+  fun checkSolutionImage(repository: String, tag: String) {
+    val images = getRepositoryTagList(repository)
+    if (!doesTheTagExist(images, tag)) {
+      throw CsmResourceNotFoundException(repository + ":" + tag)
+    }
+  }
+
+  fun doesTheTagExist(images: String, tag: String): Boolean {
+    val tags: JSONArray = JSONObject(images).get("tags") as JSONArray
```

Comment:
What happens when images is not a "json" string or does not have a "tags" property?

Reply:
The latest tag is applied by default to reference an image when you run Docker commands without specifying the tag value